org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/LSPSAXParser.java

```diff
@@ -151,11 +132,79 @@ public void doctypeDecl(String rootElement, String publicId, String systemId, Au
 						fillEntities(grammar, entityManager);
 					}
 				}
+			} catch (Exception e) {
+				reporter.reportError(locator, XMLModelMessageFormatter.XML_MODEL_DOMAIN,
+						DTDErrorCode.dtd_not_found.getCode(), new Object[] { null, eid },
+						XMLErrorReporter.SEVERITY_ERROR, e);
 			}
+
+			// Expand the system ID of the DTD and resolve it.
+			/*try {
+				String expandedSystemId = getExpandedSystemId(grammarDesc, entityManager);
+				if (!isDTDExists(expandedSystemId)) {
+					// The declared DTD doesn't exist
+					// <!DOCTYPE root-element SYSTEM "./dtd-doesnt-exist.dtd" []>
+					String message = MessageFormat.format(DTD_NOT_FOUND, expandedSystemId);
+					reportReferencedDTDError(message, DTDErrorCode.dtd_not_found.getCode(), DiagnosticSeverity.Error);
+				} else {
+					if (grammarPool != null) {
+						// FIX [BUG 2]
+						// DTD exists, get the DTD grammar from the cache
+
+						DTDGrammar grammar = (DTDGrammar) grammarPool.retrieveGrammar(grammarDesc);
+						if (grammar != null) {
+							// The DTD grammar is in cache, we need to fill XML entity manager with the
+							// entities declared in the cached DTD grammar
+							fillEntities(grammar, entityManager);
+						}
+					}
+				}
+			} catch (IOException e) {
+				Throwable cause = e.getCause();
+				if (cause instanceof CacheResourceException) {
+					reportReferencedDTDError((CacheResourceException) cause);
+				}
+			} catch (CacheResourceException e) {
+				reportReferencedDTDError(e);
+			}*/

```

Comment:
please don't leave commented code. Why is it commented in the 1st place?

Reply:
fixed